CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+If you want to implement a feature and are not too sure how to split it, just open an issue about your plans and ping other WarpX developers on it to chime in.
+
+Even before your work is ready to merge, it can be convenient to create a PR (so you can use Github tools to visualize your changes).
+In this case, please put the ``[WIP]`` tag (for Work-In-Progress) at the beginning of the PR title.
+
+Include a test to your PR
+"""""""""""""""""""""""""
+
+A new feature is great, a **working** new feature is even better!
+Please test your code and add your test to the automated test suite.
+It's the way to protect your work from adventurous developers.
+Instructions are given in the :ref:`testing section <developers-testing>` of our `developer's documentation <https://warpx.readthedocs.io/en/latest/developers/testing.html>`_.
+
+Include documentation to your PR
+""""""""""""""""""""""""""""""""
+
+Now, let users know about your new feature by adding it to the `WarpX documentation <https://warpx.readthedocs.io>`_.
+Our documentation uses `Sphinx <http://www.sphinx-doc.org/en/master/usage/quickstart.html>`_, and it is located in ``Docs/``.
+For instance, if you introduce a new runtime parameter in the input file, you can add it to :ref:`Docs/source/running_cpp/parameters.rst <running-cpp-parameters>`.
+If Sphinx is installed on your computer, you should be able to generate the html documentation with
+
+.. code-block:: sh
+
+   make html
+
+in ``Docs/``. Then open ``Docs/build/html/index.html`` with your favorite web browser and look
+for your changes.
+
+Once your code is ready with documentation and automated test, congratulations!
+You can create the PR (or remove the ``[WIP]`` tag if you already created it).
+Reviewers will interact with you if they have comments/questions.
+
+Style and conventions
+---------------------
+
+- For indentation, WarpX uses four spaces (no tabs)
+- The number of characters per line should be <100
+- To define a function , for e.g., ``myfunction()`` use a space between the name of the function and the paranthesis - ``myfunction ()``.
+  To call the function, the space is not required, i.e., just use ``myfunction()``.
+
+  - The reason this is beneficial is that when we do a ``git grep`` to search for ``myfunction ()``, we can clearly see the locations where ``myfunction ()`` is defined and where ``myfunction()`` is called.
+  - Also, using ``git grep "myfunction ()"`` searches for files only in the git repo, which is more efficient compared to the ``grep "myfunction ()"`` command that searches through all the files in a directory, including plotfiles for example.
+- It is recommended that style changes are not included in the PR where new code is added.
+  This is to avoid any errors that may be introduced in a PR just to do style change.
+- Some text editors automatically modify the files you open.
+  We recommend to turn on to remove trailing spaces and replace Tabs with 4 spaces.
```

Comment:
I would put these 2 lines as a note afterwards or before, if you choose to mention the `.github/workflow` files.

Reply:
moved down